dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-api/src/main/java/org/apache/dolphinscheduler/plugin/datasource/api/client/DruidDataSourceClient.java

```diff
@@ -0,0 +1,129 @@
+
+    private static final Logger logger = LoggerFactory.getLogger(DruidDataSourceClient.class);
+
+    public static final String COMMON_USER = "root";
+    public static final String COMMON_VALIDATION_QUERY = "select 1";
+
+    protected final BaseConnectionParam baseConnectionParam;
+    protected DruidDataSource dataSource;
+    protected JdbcTemplate jdbcTemplate;
+
+    public DruidDataSourceClient(BaseConnectionParam baseConnectionParam, DbType dbType) {
+        this.baseConnectionParam = baseConnectionParam;
+        preInit();
+        checkEnv(baseConnectionParam);
+        initClient(baseConnectionParam, dbType);
+        checkClient();
+    }
+
+    protected void preInit() {
+        logger.info("preInit in DruidDataSourceClient");
+    }
+
+    protected void checkEnv(BaseConnectionParam baseConnectionParam) {
+        checkValidationQuery(baseConnectionParam);
+        checkUser(baseConnectionParam);
+    }
+
+    /**
+     * HikariCP does not support DM, druid supports DM
+     * @param baseConnectionParam
+     * @param dbType
+     */
+    protected void initClient(BaseConnectionParam baseConnectionParam, DbType dbType) {
+        this.dataSource = JDBCDataSourceProvider.createDruidDataSource(baseConnectionParam, dbType);
+        this.jdbcTemplate = new JdbcTemplate(dataSource);
+    }
+
+    protected void checkUser(BaseConnectionParam baseConnectionParam) {
+        if (StringUtils.isBlank(baseConnectionParam.getUser())) {
+            setDefaultUsername(baseConnectionParam);
+        }
+    }
+
+    protected void setDefaultUsername(BaseConnectionParam baseConnectionParam) {
+        baseConnectionParam.setUser(COMMON_USER);
+    }
+
+    protected void checkValidationQuery(BaseConnectionParam baseConnectionParam) {
+        if (StringUtils.isBlank(baseConnectionParam.getValidationQuery())) {
+            setDefaultValidationQuery(baseConnectionParam);
+        }
+    }
+
+    protected void setDefaultValidationQuery(BaseConnectionParam baseConnectionParam) {
+        baseConnectionParam.setValidationQuery(COMMON_VALIDATION_QUERY);
+    }
+
+    @Override
+    public void checkClient() {
+        // Checking data source client
+        Stopwatch stopwatch = Stopwatch.createStarted();
+        try {
+            this.jdbcTemplate.execute(this.baseConnectionParam.getValidationQuery());
+        } catch (Exception e) {
+            throw new RuntimeException("JDBC connect failed", e);
+        } finally {
+            logger.info("Time to execute check jdbc client with sql {} for {} ms ",
+                    this.baseConnectionParam.getValidationQuery(), stopwatch.elapsed(TimeUnit.MILLISECONDS));
+        }
+    }
+
+    @Override
+    public Connection getConnection() {
+        try {
+            return this.dataSource.getConnection();
+        } catch (SQLException e) {
+            logger.error("get druidDataSource Connection fail SQLException: {}", e.getMessage(), e);
```

Comment:
```suggestion
            logger.error("get druidDataSource Connection fail SQLException: ", e);
```

Reply:
ok, I will modify it according to your suggestion